### PR TITLE
s3_scrubber: remove atty dependency

### PR DESCRIPTION
**Pull request by jcsp**
## Problem

- https://github.com/neondatabase/neon/security/dependabot/28

## Summary of changes

Remove atty, and remove the `with_ansi` arg to scrubber's stdout logger.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: s3_scrubber/Cargo.toml [hunk 1]
```diff
@@ -34,6 +34,4 @@ pageserver = {path="../pageserver"}
 tracing.workspace = true
 tracing-subscriber.workspace = true
 clap.workspace = true
-
-atty = "0.2"
 tracing-appender = "0.2"
```
Comment (koivunej) [inline s3_scrubber/Cargo.toml:37]: Could you fix this to a text file, add that missing newline in the end? I don't think I can suggest that.

your $EDITOR probably has a setting for producing only such files.

**Comment by koivunej** [comment]
Earlier PR was #5141.

File: s3_scrubber/Cargo.toml (hunk 1 above)
Comment (jcsp) [reply s3_scrubber/Cargo.toml]: Reformatted this file

File: s3_scrubber/Cargo.toml [hunk 2]
```diff
@@ -22,18 +24,21 @@ serde_json.workspace = true
 serde_with.workspace = true
 workspace_hack.workspace = true
 utils.workspace = true
-
-tokio = { workspace = true, features = ["macros", "rt-multi-thread"] }
-chrono = { workspace = true, default-features = false, features = ["clock", "serde"] }
```
Comment (arpad-m) [inline s3_scrubber/Cargo.toml:27]: I like the old style of fitting onto one line. Is there some way to preserve it? Also, having formatting changes as well as semantic ones in the same PR makes reviews harder.

File: s3_scrubber/src/lib.rs [hunk 3]
```diff
@@ -140,7 +140,6 @@ pub fn init_logging(binary_name: &str, dry_run: bool, node_kind: &str) -> Worker
         .with_writer(file_writer);
     let stdout_logs = fmt::Layer::new()
         .with_target(false)
-        .with_ansi(atty::is(atty::Stream::Stdout))
```
Comment (arpad-m) [inline s3_scrubber/src/lib.rs:143]: Could we maybe still do terminal detection e.g. by using the [`IsTerminal`](https://doc.rust-lang.org/stable/std/io/trait.IsTerminal.html) trait?

Comment (jcsp) [reply s3_scrubber/src/lib.rs:143]: perhaps, but it's easy to add that back in if we decide that we care about having colored output from the scrubber (personally I'm always piping output to a file)

File: s3_scrubber/Cargo.toml (hunk 2 above)
Comment (jcsp) [reply s3_scrubber/Cargo.toml:27]: I suggest we ignore whitespace in these files until we standardize on a linter -- the style here is just what the most popular VSCode toml plugin does.

Comment (jcsp) [reply s3_scrubber/Cargo.toml:27]: I've just reverted this and added a newline to the file (Joonas asked for that in an earlier review).

File: s3_scrubber/src/lib.rs (hunk 3 above)
Comment (jcsp) [reply s3_scrubber/src/lib.rs:143]: This was a one liner so I just added IsTerminal.

**Review by arpad-m — APPROVED** [review]

**Pull request merged**